spec/controllers/dogs_controller_spec.rb

```diff
@@ -0,0 +1,116 @@
+require 'spec_helper'
+
+describe DogsController, type: :controller do
+
+  shared_examples 'an OK response with 1 dog' do
+    it 'responds successfully' do
+      expect(response).to be_success
+    end
+
+    it 'contains 1 dog' do
+      last_json = JSON.parse(response.body)
+      expect(last_json.length).to eq(1)
+    end
+
+    it 'contains found dog' do
+      expect([found_dog].to_json).to be_json_eql(response.body)
+    end
+  end
+
+  describe 'GET #index' do
+    context 'user is logged in' do
+      let!(:found_dog) { create(:dog, tracking_id: 1) }
+      let!(:other_dog) { create(:dog, tracking_id: 100) }
```

Comment:
Ah duh that's a good point. Wasn't entirely obvious to me when reading this. 

Which is why I like my 4 phases of a test as close to each other as possible :D